pilot/cmd/pilot-discovery/main.go

```diff
@@ -120,6 +120,8 @@ func init() {
 	serverArgs.MCPCredentialOptions.AttachCobraFlags(discoveryCmd)
 	discoveryCmd.PersistentFlags().IntVar(&serverArgs.MCPMaxMessageSize, "mcpMaxMsgSize", bootstrap.DefaultMCPMaxMsgSize,
 		"Max message size received by MCP's grpc client")
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.IncrementalEDS, "incrementalEDS", false,
```

Comment:
This is going to change in my next commit. It might be a list of galleys which support incremental MCP, but not 100% sure since still thinking about it.

Reply:
Let's not even bother with this - I can't imagine any reason to have a MCP endpoint source that is not incremental. 

And we already have a way to configure multiple MCP sources - we may need to separate the single 'config' source
from the list of 'endpoints' sources, but that's in mesh config, no need for flags in pilot.

Reply:
If we are planning to split the coredatamodel into two controllers, we need a way of distinguishing which Galleys are configured for SyntheticServiceEntry controller.

Reply:
@costinm are you saying that all MCP connections should be incremental? Or just the ones for ServiceEntries? Is the existing ServiceEntry logic (custom CRDs) considered an alpha-level feature? If not I'm not sure we can just switch over to incremental.

In addition, this flag could potentially allow us to configure SSE generation for "this" cluster ... where we disable the k8s layer entirely.

Reply:
I am saying that 'incremental EDS' is a feature we support already on XDS side. 

We may switch all MCP connections to incremental at some point - main blocker is having enough
stability testing. That's a different discussion and feature.

Same for 'this' cluster - we plan to disable k8s layer entirely, but it's extremely unlikely this will be in 1.2,
it'll need quite a few iterations and a lot of stability. We are starting with remote clusters - where we already 
have a working solution to reduce risks.